### PR TITLE
PILOT-1292 Use init container for running migrations

### PR DESCRIPTION
**Pull request by erikdvlp**
## Summary

- Change Docker files to run Alembic migrations in their own container / build stage.
- Move local default Postgres credentials from `docker-compose.yaml` to `config.py`.
- Update readme with new template and migration instructions.

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1292

## Type of Change

- [x] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [ ] Yes
- [x] No

## Test Directions

- Run the service locally and ensure that migrations run. They are running in Actions.
- Compare to the [project service](https://github.com/PilotDataPlatform/project) for consistency of approach.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -49,7 +46,18 @@ services:
...
-      - db
+      db:
+        condition: service_healthy
+
+  alembic:
+    build:
+      context: .
+      target: alembic-image
+    environment:
+      - SQLALCHEMY_DATABASE_URI=${SQLALCHEMY_DATABASE_URI:-postgresql://postgres:postgres@metadata_db_1:5432/metadata}
```
Comment (vmoshynskyi) [inline docker-compose.yaml:57]: As I understand it won't work in most cases with `metadata_db_1`:
```suggestion
      - SQLALCHEMY_DATABASE_URI=${SQLALCHEMY_DATABASE_URI:-postgresql://postgres:postgres@db:5432/metadata}
```

File: Dockerfile [hunk 2]
```diff
@@ -22,5 +22,9 @@ RUN pip install --no-cache-dir poetry==1.1.12
 RUN poetry config virtualenvs.create false
 RUN poetry install --no-dev --no-root --no-interaction
 
-RUN chmod +x uvicorn_starter.sh
-CMD ["./uvicorn_starter.sh"]
```
Comment (vmoshynskyi) [inline Dockerfile:26]: What output docker image we will use? And what entrypoint it will have if we don't use `uvicorn_starter.sh` anymore?

**Review by vmoshynskyi — CHANGES_REQUESTED** [review]
Please check my comments.

File: .github/workflows/test_docker_compose.yml [hunk 3]
```diff
@@ -30,25 +30,20 @@ jobs:
...
-          echo "OPSDB_UTILILY_PASSWORD=postgres" >> $GITHUB_ENV
-          echo "OPSDB_UTILILY_HOST=db" >> $GITHUB_ENV
-          echo "OPSDB_UTILILY_NAME=metadata" >> $GITHUB_ENV
           [ ! -f .env ] && touch .env
           [ -d local_config/pgadmin/sessions ] && sudo chmod 777 -R -f local_config/pgadmin/sessions
           sudo chmod 777 -R -f local_config/pgadmin/sessions
-          docker build -t web .
-          docker network create pilot-network
           docker-compose -f docker-compose.yaml down -v
+          docker network create pilot-network
```
[the remark this answers is not in the document]
Comment (erikdvlp) [reply .github/workflows/test_docker_compose.yml]: Removed.

File: docker-compose.yaml (hunk 1 above)
Comment (erikdvlp) [reply docker-compose.yaml]: I modified Alembic's config to not use an environment variable for its database connection.

File: Dockerfile (hunk 2 above)
Comment (erikdvlp) [reply Dockerfile:26]: I updated the Docker files to use `metadata-image` with an entrypoint.

File: .github/workflows/test_docker_compose.yml [hunk 4]
```diff
@@ -30,25 +30,20 @@ jobs:
...
-          pwd
-          hostname
-          docker-compose exec -T web pip install --user poetry==1.1.12
-          docker-compose exec -T web poetry config virtualenvs.in-project false
-          docker-compose exec -T web poetry install --no-root --no-interaction
-          docker-compose exec -T web poetry run pytest --verbose -c tests/pytest.ini
+          docker exec metadata_web_1 pip install --user poetry==1.1.12
+          docker exec metadata_web_1 poetry config virtualenvs.in-project false
+          docker exec metadata_web_1 poetry install --no-root --no-interaction
+          docker exec metadata_web_1 poetry run pytest --verbose -c tests/pytest.ini
```
[the remark this answers is not in the document]
Comment (erikdvlp) [reply .github/workflows/test_docker_compose.yml]: This whole thing should be reworked, but in a separate task.

File: Jenkinsfile [hunk 5]
```diff
@@ -18,57 +18,27 @@ pipeline {
...
       }
     }
 
     stage('DEV Remove image') {
       when {branch "develop"}
-      steps{
-        sh "docker rmi $imagename:$commit-CAC"
+      steps {
+            sh 'docker rmi $imagename:alembic-$commit'
+            sh 'docker rmi $imagename:project-$commit'
```
Comment (vmoshynskyi) [inline Jenkinsfile:41]: Also here:
```suggestion
            sh 'docker rmi $imagename:alembic-$commit-CAC'
            sh 'docker rmi $imagename:metadata-$commit-CAC'
```

**Review by vmoshynskyi — CHANGES_REQUESTED** [review]
Everything is good. A couple of final edits.

File: Jenkinsfile [hunk 6]
```diff
@@ -18,57 +18,27 @@ pipeline {
...
     stage('DEV Build and push image') {
       when {branch "develop"}
       steps {
         script {
-          docker.withRegistry('https://ghcr.io', registryCredential) {
-              customImage = docker.build("$imagename:$commit-CAC")
-              customImage.push()
-          }
+            docker.withRegistry('https://ghcr.io', registryCredential) {
+                customImage = docker.build('$imagename:alembic-$commit', '--target alembic-image .')
```
[the remark this answers is not in the document]
Comment (erikdvlp) [reply Jenkinsfile]: You're right, thanks for catching these.

**Pull request merged**